eng/tools/compare-packages.ps1

```diff
@@ -0,0 +1,57 @@
+param (
+  $pathToMasterPkg,
+  $pathToCurrentPkg,
+  $dailyDevBuildNo,
+  $pathForDiffFile
+)
+
+function ExtractTGZPackages($pathToPkg) {
+  $parentExtractDir = Join-Path $pathToPkg "all-contents"
+  Write-Host "mkdir $parentExtractDir"
+  mkdir $parentExtractDir
+
+  foreach ($p in $(dir $pathToPkg -r -i *.tgz)) {
+    if($p.BaseName -match "(?<name>.*?)(-\d+\.\d+\.\d+.*)") {
+      $extractDir = Join-Path $parentExtractDir $matches["name"]
+      mkdir "$extractDir"
+      pushd "$extractDir"
+      tar -xzf $p.FullName
+      popd
+    }
+    else {
+      write-error "Package name $($p.BaseName) doesn't match the expected format [<name>-<version>]!"
+    }
+  }
+}
+
+try {
+  $extractMasterDir = Join-Path $pathToMasterPkg "all-contents"
+  ExtractTGZPackages($pathToMasterPkg)
+
+  if($LastExitCode -ne 0) {
```

Comment:
I assume the only command you are looking for the lastexitcode from is the tar command. It is probably more appropriate to put this error check directly after that command instead of after the function. Putting it here means you know too much about the inner workings of the ExtractTGZPackages function plus in the case of multiple packages this is really only checking the last tar command and not each individual one so if one fails in the middle it might not be caught.

Reply:
done